soc: nordic: common: Fix HAS_HW_NRF_NFCT condition

nordic-krch opened the pull request:
Include nordic,nrf-nfct-v2 compatible in the option.

nordic-krch committed:
soc: nordic: common: Fix HAS_HW_NRF_NFCT condition

Include nordic,nrf-nfct-v2 compatible in the option.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

anangl commented on dts/vendor/nordic/nrf54h20.dtsi:
```diff
@@ -920,7 +920,7 @@
 
 			nfct: nfct@985000 {
-				compatible = "nordic,nrf-nfct";
+				compatible = "nordic,nrf-nfct-v2";
```
It would be better to do this in a separate commit, as a follow-up to 7249fac80d60b84d2ba9c2a1de6d42d8b3d9bd0c.

nordic-krch replied:
split done.

anangl commented on soc/nordic/nrf54h/soc.c:
```diff
@@ -153,7 +153,9 @@ void soc_early_init_hook(void)
-	if (DT_NODE_HAS_STATUS(DT_NODELABEL(nfct), disabled) &&
+	if (((IS_ENABLED(CONFIG_SOC_NRF54H20_CPUAPP) &&
+	      DT_NODE_HAS_STATUS(DT_NODELABEL(nfct), disabled)) ||
+	     DT_NODE_HAS_STATUS(DT_NODELABEL(nfct), reserved)) &&
```
I think it should be mentioned somewhere (maybe in the `nordic,nrf-nfct-v2` binding, in the `nfct-pins-as-gpios` property description) that "reserved" should be used as the node status, so that the NFCT peripheral is assigned to a given core.

nordic-krch committed:
soc: nordic: nrf54h: Allow using NFCT pins as gpios on cpurad

NFCT is by default assigned to application so DT node does
not need to be enabled or reserved in DT to have access to
NFCT registers. On cpurad NFCT must be reserved to enable
register access and then antenna pins can be configured as
gpios.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
dts: vendor: nordic: Fix nfct compatible

Use nordic,nrf-nfct-v2 compatible in the main nrf54h20 devicetree
description.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
dts: bindings: nordic-nrf-nfct-v2: Extend description

Add information about nfct node status. For cpuapp status is irrelevant
because NFCT is by default assigned to cpuapp but for cpurad node needs
to have reserved status.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>
